yb-voyager/src/anon/sql_anonymizer.go

```diff
@@ -1823,6 +2108,102 @@ func (a *SqlAnonymizer) anonymizeRangeVarNode(rv *pg_query.RangeVar, finalPrefix
 	return nil
 }
 
+// anonymizeRangeTypeParameters handles all parameters for CREATE TYPE ... AS RANGE statements
+/*
+Refer: https://www.postgresql.org/docs/current/sql-createtype.html
+
+CREATE TYPE name AS RANGE (
+    SUBTYPE = subtype
+    [ , SUBTYPE_OPCLASS = subtype_operator_class ]
+    [ , COLLATION = collation ]
+    [ , CANONICAL = canonical_function ]
+    [ , SUBTYPE_DIFF = subtype_diff_function ]
+    [ , MULTIRANGE_TYPE_NAME = multirange_type_name ]
+)
+*/
+func (a *SqlAnonymizer) anonymizeRangeTypeParameters(params []*pg_query.Node) error {
+	if params == nil {
+		return nil
+	}
+
+	for _, defElem := range params {
+		defElemNode := defElem.GetDefElem()
+		if defElemNode == nil || defElemNode.Defname == "" {
+			continue
+		}
+
+		switch defElemNode.Defname {
+		case "subtype":
+			// For range types, subtype should be anonymized as TYPE
+			if defElemNode.Arg == nil {
+				continue
+			}
+			typeName := defElemNode.Arg.GetTypeName()
+			if typeName == nil || IsBuiltinType(typeName) {
+				continue
+			}
+			if err := a.anonymizeStringNodes(typeName.Names, TYPE_KIND_PREFIX); err != nil {
+				return fmt.Errorf("anon range subtype: %w", err)
+			}
+
+		case "subtype_opclass":
+			// Handle operator class names
+			if defElemNode.Arg == nil {
+				continue
+			}
+			typeName := defElemNode.Arg.GetTypeName()
+			if err := a.anonymizeStringNodes(typeName.Names, OPCLASS_KIND_PREFIX); err != nil {
+				return fmt.Errorf("anon range subtype_opclass: %w", err)
+			}
+
+		case "collation":
+			// Handle collation names
+			if defElemNode.Arg == nil {
+				continue
+			}
+			typeName := defElemNode.Arg.GetTypeName()
+			if err := a.anonymizeStringNodes(typeName.Names, COLLATION_KIND_PREFIX); err != nil {
+				return fmt.Errorf("anon range collation: %w", err)
+			}
+
+		case "canonical":
+			// Handle canonical function names
+			if defElemNode.Arg == nil {
+				continue
+			}
+			typeName := defElemNode.Arg.GetTypeName()
+			if err := a.anonymizeStringNodes(typeName.Names, FUNCTION_KIND_PREFIX); err != nil {
+				return fmt.Errorf("anon range canonical: %w", err)
+			}
+
+		case "subtype_diff":
+			// Handle subtype_diff function names
+			if defElemNode.Arg == nil {
+				continue
+			}
+			typeName := defElemNode.Arg.GetTypeName()
+			if err := a.anonymizeStringNodes(typeName.Names, FUNCTION_KIND_PREFIX); err != nil {
+				return fmt.Errorf("anon range subtype_diff: %w", err)
+			}
+
+		case "multirange_type_name":
+			// Handle multirange type names
+			if defElemNode.Arg == nil {
+				continue
+			}
+			typeName := defElemNode.Arg.GetTypeName()
+			if typeName == nil || IsBuiltinType(typeName) {
+				continue
+			}
+			if err := a.anonymizeStringNodes(typeName.Names, TYPE_KIND_PREFIX); err != nil {
+				return fmt.Errorf("anon range multirange_type_name: %w", err)
+			}
+		}
+	}
```

Comment:
Here as well, it looks like the difference in all these cases is just of prefix so we just have the logic of anonymizing it at the end and only deciding the prefix in these